Zend/zend_language_parser.y

```diff
@@ -1075,7 +1102,13 @@ new_variable:
 ;
 
 member_name:
-		T_STRING		{ $$ = $1; }
+		identifier { $$ = $1; }
+	|	'{' expr '}'	{ $$ = $2; }
+	|	simple_variable	{ $$ = zend_ast_create(ZEND_AST_VAR, $1); }
+;
+
+property_name:
```

Comment:
BTW, I found a way to uncomment  https://github.com/php/php-src/pull/1158/files#diff-7eff82c2c5b45db512a9dc49fb990bb8R280 and have these words semi reserved too, but that would require people using methods that clash with method modifiers name to use the verbose syntax for trait aliasing. Ex:


```php
trait TraitA {
   private function private(){}
}

Class A {
    use TraitA {
        private as protected protected;
    }
}
```

I'm leaving this out of the patch though, as it has to be compatible with the older one.